### DISCUSSION TITLE
Why not interrupting current command on statement close

### DISCUSSION THREAD

**Discussion by RLhereec** [microsoft/mssql-jdbc#2039] [Q&A] [answered]
Hi guys,
I am surprised that the current command is not interrupted on SqlServerStatement close method.
I wonder what is the rationale behind this choice ? 
Regards,
Rémy

**Comment by David-Engel** [comment, marked as answer]
Canceling (interrupting) a command sends a signal to the server to stop processing the current execution. Current close behavior allows server-side processing to complete (by waiting and consuming all results) before closing the connection. If we changed to cancel on close, (1) the behavior would change and (2) the behavior would be inconsistent/indeterminate server-side results of any in-process executions. (There could be any number of T-SQL commands/stored procedures/inserts/updates running in a single execution and the cancel could happen anywhere inside/during that execution.)

Regards,
David
